github/Requester.py

```diff
@@ -999,19 +1061,24 @@ def __requestRaw(
         url: str,
         requestHeaders: Dict[str, str],
         input: Optional[Any],
-    ) -> Tuple[int, Dict[str, Any], str]:
+        stream: bool = False,
+    ) -> Tuple[int, Dict[str, Any], Union[str, requests.Response]]:
         self.__deferRequest(verb)
 
         try:
             original_cnx = cnx
             if cnx is None:
                 cnx = self.__createConnection()
-            cnx.request(verb, url, input, requestHeaders)
-            response = cnx.getresponse()
-
-            status = response.status
-            responseHeaders = {k.lower(): v for k, v in response.getheaders()}
-            output = response.read()
+            if stream:
+                response = cnx.session.get(url, stream=True, headers=requestHeaders)
```

Comment:
The `True` branch assumes `verb` to be `get`.

Reply:
Is that a question, or a statement?
Since the code in question is in `__requestRaw()` I suppose it could be reasonable to assume that it might be useful for some other verb.  But, all the other code in this library assumes that the returned content is text, so it would be a much bigger task to review it all and figure out what other changes would be necessary to make it work with other verbs and other encoding.

I also looked at the GitHub API and did not find anything other than Release Assets that can return a binary stream.  Updating a Release Asset does require using the `POST` verb, but that does not return a stream.

This is the reason I hard-coded the verb.  Release Assets only need `GET` and no additional data needs to be provided.

Reply:
This is now no longer hard-coding GET at this point.